src/AdapterPluginManagerFactory.php

```diff
@@ -5,29 +5,25 @@
 namespace Laminas\Paginator;
 
 use Laminas\ServiceManager\Config;
-use Laminas\ServiceManager\FactoryInterface;
-use Laminas\ServiceManager\ServiceLocatorInterface;
+use Laminas\ServiceManager\ServiceManager;
 use Psr\Container\ContainerInterface;
 
 use function is_array;
 
-class AdapterPluginManagerFactory implements FactoryInterface
+/**
+ * @psalm-import-type ServiceManagerConfiguration from ServiceManager
+ * @final
+ */
+class AdapterPluginManagerFactory
```

Comment:
Hmm, not super convinced about these changes (for BC purposes).

It is true that we added `@final`, but we still dropped methods and whatnot :| This is clearly a BC breaking change.

Reply:
Thanks!